sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -1572,7 +1572,7 @@ internal virtual async ValueTask<Response<BlobDownloadStreamingResult>> Download
                         ClientConfiguration.Pipeline.ResponseClassifier,
                         Constants.MaxReliabilityRetries);
 
-                    stream = stream.WithNoDispose().WithProgress(progressHandler);
```

Comment:
Very curious to know if this is okay to do. My understanding is that we use this extension specifically for upload methods to ensure we don't dispose a stream that was handed to us. But if we do it on download like we do here, then the caller can never possibly dispose the network stream properly. If there is something important we are using this to prevent disposal of, we need to find a different mechanism, but I don't think there is one.

Reply:
This change makes sense to me that we only want to make the stream non-disposable for uploads and not for downloads. I think this change should be safe but I'm also wondering why we did this in the first place? Maybe just incorrectly copying from the upload method?

Reply:
Yeah I was also looking to remove this when I was considering a fix for this issue. But I didn't know if the "NoDispose" was necessary for features like progress handling.

But it looks like the existing tests are fine so this might be a copy and paste error from the upload implementation I'm not sure.